sktime/forecasting/compose/_pipeline.py

```diff
@@ -1448,3 +1450,192 @@ def get_test_params(cls, parameter_set="default"):
+    >>> pipe = ForecastingPipeline(
+    ...     [
+    ...         ("boxcox", BoxCoxTransformer()),
+    ...         ("exp", ExponentTransformer(3)),
+    ...         ("naive", NaiveForecaster()),
+    ...     ]
+    ... )
+    >>> # this results in the pipeline with sequence "exp", "boxcox", "naive"
+    >>> permuted = Permute(pipe, ["exp", "boxcox", "naive"])
+    >>> pipe = pipe.fit(y, fh=fh)
+    >>> y_pred = pipe.predict()
+
+    The permuter is useful in combination with grid search (toy example):
+    >>> from sktime.datasets import load_shampoo_sales
+    >>> from sktime.forecasting.model_selection import (
+    ...     ExpandingWindowSplitter,
+    ...     ForecastingGridSearchCV,
+    ... )
+    >>> fh = [1,2,3]
+    >>> cv = ExpandingWindowSplitter(
+    ...     start_with_window=True,
+    ...     fh=fh)
+    >>> forecaster = NaiveForecaster()
+    >>> # check which of the two sequences of transformers is better
+    >>> param_grid = {
+    ...     "permutation" : [["boxcox", "exp", "naive"], ["exp", "boxcox", "naive"]]
+    ... }
+    >>> gscv = ForecastingGridSearchCV(
+    ...     forecaster=forecaster,
+    ...     param_grid=param_grid,
+    ...     cv=cv)
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "y_inner_mtype": ALL_TIME_SERIES_MTYPES,
+        "X_inner_mtype": ALL_TIME_SERIES_MTYPES,
+        "ignores-exogeneous-X": False,
+        "requires-fh-in-fit": False,
+        "handles-missing-data": True,
+        "capability:pred_int": True,
+        "X-y-must-have-same-index": False,
+    }
+
+    _delegate_name = "estimator_"
+
+    def __init__(self, estimator, permutation, steps_arg="steps"):
+        self.estimator = estimator
+        self.permutation = permutation
+        self.steps_arg = steps_arg
+
+        super(Permute, self).__init__()
+        tags_to_clone = [
+            "ignores-exogeneous-X",  # does estimator ignore the exogeneous X?
+            "capability:pred_int",  # can the estimator produce prediction intervals?
+            "requires-fh-in-fit",  # is forecasting horizon already required in fit?
+            "enforce_index_type",  # index type that needs to be enforced in X/y
+            "fit_is_empty",
+        ]
+
+        self.clone_tags(self.estimator, tags_to_clone)
+
+        self._set_permuted_estimator()
+
+    def _set_permuted_estimator(self):
+        """Set self.estimator_ based on permutation arg."""
+        estimator = self.estimator
+        permutation = self.permutation
+        steps_arg = self.steps_arg
+
+        self.estimator_ = estimator.clone()
+
+        if permutation is not None:
+
+            inner_estimators = getattr(estimator, steps_arg)
+            estimator_tuples = self._get_estimator_tuples(inner_estimators)
+
+            estimator_dict = {x[0]: x[1] for x in estimator_tuples}
+
+            # check that all permutation are str values
+            if not all(isinstance(item, str) for item in permutation):
```

Comment:
might raise unexpected errors if permutation is not a list

Reply:
this just follows 1:1 the logic of @aiwalter's PR.
I agree though we may want to have a more user friendly check though.

Reply:
added the check